client/src/main/java/io/pravega/client/segment/impl/SegmentInputStreamImpl.java

```diff
@@ -119,8 +122,11 @@ public ByteBuffer read(long timeout) throws EndOfSegmentException {
         }
     }
 
-    private ByteBuffer readEventData(long timeout) throws EndOfSegmentException {
+    private ByteBuffer readEventData(long timeout) throws EndOfSegmentException, SegmentTruncatedException {
         fillBuffer();
+        if (receivedTruncated) {
```

Comment:
Looks like `receivedTruncated ` is never set to true. 

`io.pravega.client.segment.impl.SegmentInputStreamImpl#handleRequest` now throws a SegmentTruncatedException wrapped inside CompletedException when AsyncSegmentInputStreamImpl#ResponseProcessor receives a NoSuchSegment/SegmentTruncated.

This implies `io.pravega.client.segment.impl.SegmentInputStreamImpl#read` throws a `CompletionException`, which should not be the case.

Reply:
Done